feat: support setting ICE ufrag and pwd

achingbrain opened the pull request:
https://github.com/paullouisageneau/libjuice/pull/243 allows setting the ICE ufrag and pwd fields instead of generating random ones every time.

https://github.com/paullouisageneau/libdatachannel/pull/1201 exposes config keys to allow setting the fields in libjuice from libdatachannel.

This PR allows setting the fields in libdatachannel from the PeerConnection constructor.

It will require the two PRs above being merged an shipped before this is ready for merging.

Refs: https://github.com/paullouisageneau/libdatachannel/issues/1166

murat-dogan commented:
Thanks for the PR.
It would be good if you also add these changes here;
* https://github.com/murat-dogan/node-datachannel/blob/a862c23ca1f690ff83c592586e612b7189928c56/lib/index.d.ts#L280
* https://github.com/murat-dogan/node-datachannel/blob/master/API.md

murat-dogan commented:
Hello,
Thanks for the work!

When that could be ready for merging?

unicomp21 commented:
@murat-dogan @achingbrain what remains before we can merge?

achingbrain commented:
For the functionality to be usable this PR needs to be merged/released as well - https://github.com/paullouisageneau/libjuice/pull/248 - then this one: https://github.com/paullouisageneau/libdatachannel/pull/1211 - then we can expose it here.

unicomp21 commented:
given the delay, would it make sense to consider forking in the meantime?

unicomp21 commented:
https://github.com/murat-dogan/node-datachannel/pull/308

murat-dogan commented:
Hi all,

I think forking the lib is not feasible.
Let’s ping @paullouisageneau to see if there’s a problem to be solved for PRs @achingbrain  mentioned.

To merge only fingerprint feature is something @achingbrain should decide, while he worked on this PR already. If this will be the case also this PR must be re-worked.

paullouisageneau commented:
I'll do another pass on https://github.com/paullouisageneau/libjuice/pull/248

This issue was closed but as I understand it is still blocking, as the ICE ufrag and pwd can't be applied from the polyfill: https://github.com/paullouisageneau/libdatachannel/issues/1218

murat-dogan commented:
Is there something that I can do @paullouisageneau ?

paullouisageneau commented:
I've merged https://github.com/paullouisageneau/libjuice/pull/248, next steps are merging https://github.com/paullouisageneau/libjuice/pull/248 and [fixing the polyfill](https://github.com/paullouisageneau/libdatachannel/issues/1218).

achingbrain commented:
I've updated this branch and fixed the merge conflicts.

I've also updated https://github.com/paullouisageneau/libdatachannel/pull/1211 to use the API landed in https://github.com/paullouisageneau/libjuice/pull/248 though it's still waiting for a libjuice release.  Optionally https://github.com/paullouisageneau/libjuice/pull/291 exposes a the slightly more explicit API - it could be merged and https://github.com/paullouisageneau/libdatachannel/pull/1211 updated again or closed - it's up to @paullouisageneau.

achingbrain commented:
The unhandled STUN callback is in now but it needs:

- [ ] https://github.com/paullouisageneau/libjuice/pull/292
- [ ] https://github.com/paullouisageneau/libdatachannel/pull/1211

achingbrain commented:
Just checking in here.

I'm still waiting for @paullouisageneau to do a libjuice release that includes https://github.com/paullouisageneau/libjuice/pull/292 which can then be rolled up to https://github.com/paullouisageneau/libdatachannel/pull/1211

paullouisageneau commented:
@achingbrain Done, I've released libjuice v1.6.0 with github.com/paullouisageneau/libjuice/pull/292

achingbrain commented:
@paullouisageneau awesome, thanks!  I've updated [libdatachannel#1211](https://github.com/paullouisageneau/libdatachannel/pull/1211) to use the new `v1.6.0` release.

achingbrain commented:
Checking in again.

Currently waiting for @paullouisageneau to do a libdatachannel release that includes https://github.com/paullouisageneau/libdatachannel/pull/1211 - then this PR will be ready to go.

achingbrain commented:
@murat-dogan since you've switched to the HEAD of libdatachannel this PR should be ready now.
